pkg/quota_management/account.go

```diff
@@ -1,20 +1,69 @@
 package quota_management
 
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+)
+
 type Account struct {
-	Username            string `yaml:"username"`
-	MaxAllowedInstances int    `yaml:"max_allowed_instances"`
+	Username            string    `yaml:"username"`
+	MaxAllowedInstances int       `yaml:"max_allowed_instances"`
+	GrantedQuota        QuotaList `yaml:"granted_quota,omitempty"`
 }
 
-func (account Account) IsInstanceCountWithinLimit(count int) bool {
-	return count <= account.GetMaxAllowedInstances()
+var _ QuotaManagementListItem = &Account{}
+
+func (account Account) IsInstanceCountWithinLimit(instanceTypeID string, billingModelName string, count int) bool {
+	if billingModelName == "" {
+		billingModelName = "standard"
+	}
+	return count <= account.GetMaxAllowedInstances(instanceTypeID, billingModelName)
 }
 
-func (account Account) GetMaxAllowedInstances() int {
-	if account.MaxAllowedInstances <= 0 {
-		return MaxAllowedInstances
+func (account Account) GetMaxAllowedInstances(instanceTypeID string, billingModelName string) int {
+	if billingModelName == "" {
+		billingModelName = "standard"
+	}
+
+	bm, ok := account.getBillingModel(instanceTypeID, billingModelName)
+
+	if !ok {
+		return 0
 	}
 
-	return account.MaxAllowedInstances
+	if bm.Allowed <= 0 {
+		if account.MaxAllowedInstances <= 0 {
+			return MaxAllowedInstances
+		}
+		return account.MaxAllowedInstances
+	}
+
+	return bm.Allowed
+}
+
+func (account Account) GetGrantedQuota() QuotaList {
+	if len(account.GrantedQuota) == 0 {
+		return defaultInstanceTypes
```

Comment:
Same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1368/files#r1015633583

Reply:
This is for compatibility with the previous configuration file: if no quota is specified, it defaults to 'STANDARD' to keep working as before.